index.js

```diff
@@ -121,7 +121,7 @@ module.exports = {
 		'no-unmodified-loop-condition':  'error',
 		'no-unneeded-ternary':           'error',
 		'no-unused-expressions':         [ 'warn', { 'allowShortCircuit': true, 'allowTernary': true } ],
-		'no-unused-vars':                'warn',
+		'no-unused-vars':                [ 'warn', { 'argsIgnorePattern': '^_', 'varsIgnorePattern': '^_' } ],
```

Comment:
I think this will match `const _  = require('lodash'); // unused`? Maybe we want to check for existence of characters after to avoid false negatives on unused imports of lodash/underscore?

Reply:
Hmm, that's hard because sometimes people use `_` for an unnecessary variable name (although it's better to label it)`.

I guess `^_.+`?

Reply:
I've seen the underscore prefix used quite a lot to represent private variables or functions more than I've seen them represent unused variables. If I remember correctly they're usually properties on an object/class which is being exported rather than standalone variables so this rule probably doesn't apply, but worth keeping in mind.

Reply:
Afaik, ESLint should still warn you about unused functions regardless of this change. Internal private variables would be missed, I think, but using `_*` for something unused is a standard in many languages and we should probably adopt it.

Really, the case I'm interested in is `(err, result)`. There's a bunch of places people have just removed `result`, which leads me down a rabbit hole of trying to figure out exactly _what_ I have access to. If it was `_org`, it would immediately be clear that I have access to an `org`, whilst keeping the linter happy. 

Perhaps we only set this on `argsIgnorePattern` in that case? Seems like a decent enough compromise.

Reply:
I have seen `_` used for "private" variables as @alasdairhurst  mentioned, as well as `__` (double underscore) used to mean "internal variable - please don't use or depend on this".   My opinion is that if a variable is unused or unnecessary, it should be commented out or removed.

Reply:
For variables, that's fine since people do it all over JS land due to the lack of private scoping in the language itself.

Arguments is a different story though (because they're all "private"). Underscore prefixing lends itself well to code documentation here where people are too lazy to document otherwise.